packages/core/src/tools/create-cache.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+const _ = require('lodash');
+
+const JSON = require('./create-json-tool')();
+
+const createCacheObj = (input) => {
+  const rpc = _.get(input, '_zapier.rpc');
+  const runValidationChecks = (rpc, key, value = null, ttl = null) => {
+    if (!rpc) {
+      throw new Error('rpc is not available');
+    }
+
+    if (!_.isString(key)) {
+      throw new Error('key must be a string');
+    }
+
+    if (!JSON.stringify(value)) {
```

Comment:
`JSON.stringify()` is not a reliable way to tell if a value is JSON-encodable. For example:

```
> JSON.stringify(console)
'{}'
> JSON.stringify({func: function() {}})
'{}'
```

I think we need to do something like [this](https://stackoverflow.com/questions/30579940/reliable-way-to-check-if-objects-is-serializable-in-javascript).

Suppose the function is named `isJsonEncodable()`. If `value` is a primitive type, like a number or string, `isJsonEcodable(value)` simply returns true. If `value` is a nested type, like an object or array, we need to recursively iterate each element and call `isJsonEncodable(element)`.